File: clients/PHP-client/src/CronClient.php
<?php

namespace SoliDB;

class CronClient
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    public function list(): array
    {
        $res = $this->client->sendCommand('list_cron_jobs', ['database' => $this->client->getDatabase()]);
        return $res['data'] ?? [];
    }

    public function create(string $name, string $schedule, string $scriptPath, ?array $params = null, ?bool $enabled = null, ?string $description = null): array
    {
        $args = [
            'database' => $this->client->getDatabase(),
            'name' => $name,
            'schedule' => $schedule,
            'script_path' => $scriptPath
        ];
        if ($params !== null) {
            $args['params'] = $params;
        }
        if ($enabled !== null) {
            $args['enabled'] = $enabled;
        }
        if ($description !== null) {
            $args['description'] = $description;
        }
        $res = $this->client->sendCommand('create_cron_job', $args);
        return $res['data'] ?? [];
    }

    public function get(string $cronId): array
    {
        $res = $this->client->sendCommand('get_cron_job', [
            'database' => $this->client->getDatabase(),
            'cron_id' => $cronId
        ]);
        return $res['data'] ?? [];
    }

    public function update(string $cronId, array $updates): array
    {
        $res = $this->client->sendCommand('update_cron_job', [
            'database' => $this->client->getDatabase(),
            'cron_id' => $cronId,
            'updates' => $updates
        ]);
        return $res['data'] ?? [];
    }

    public function delete(string $cronId): void
    {
        $this->client->sendCommand('delete_cron_job', [
            'database' => $this->client->getDatabase(),
            'cron_id' => $cronId
        ]);
    }

    public function toggle(string $cronId, bool $enabled): void
    {
        $this->client->sendCommand('toggle_cron_job', [
            'database' => $this->client->getDatabase(),
            'cron_id' => $cronId,
            'enabled' => $enabled
        ]);
    }
}

File: clients/Ruby-client/lib/solidb/cron_client.rb
module SoliDB
  class CronClient
    def initialize(client)
      @client = client
    end

    def list
      @client.send_command("list_cron_jobs", database: @client.database) || []
    end

    def create(name:, schedule:, script_path:, params: nil, enabled: nil, description: nil)
      args = {
        database: @client.database,
        name: name,
        schedule: schedule,
        script_path: script_path
      }
      args[:params] = params if params
      args[:enabled] = enabled unless enabled.nil?
      args[:description] = description if description
      @client.send_command("create_cron_job", args)
    end

    def get(cron_id)
      @client.send_command("get_cron_job", database: @client.database, cron_id: cron_id)
    end

    def update(cron_id, updates)
      @client.send_command("update_cron_job", database: @client.database, cron_id: cron_id, updates: updates)
    end

    def delete(cron_id)
      @client.send_command("delete_cron_job", database: @client.database, cron_id: cron_id)
      nil
    end

    def toggle(cron_id, enabled)
      @client.send_command("toggle_cron_job", database: @client.database, cron_id: cron_id, enabled: enabled)
      nil
    end
  end
end
